lib/dry/operation/class_context/steps_method_prepender.rb

```diff
@@ -18,9 +37,13 @@ def included(klass)
 
         def mod
           @module ||= Module.new.tap do |mod|
-            mod.define_method(@method) do |*args, **kwargs, &block|
-              steps do
-                super(*args, **kwargs, &block)
+            module_exec(@result_handler) do |result_handler|
+              mod.define_method(@method) do |*args, **kwargs, &block|
+                steps do
+                  super(*args, **kwargs, &block)
+                end.tap do |result|
+                  result_handler.(result, self, __method__)
+                end
```

Comment:
What I would love here is we can somehow _name the individual steps themselves_ and have it passed to the result handler. That way it could do even finer-grained handling of errors.

Perhaps we could consider an optional syntax like `step :step_name, value`. Converting our example from our inline docs:

```ruby
def call(input)
  attrs = step :validate, validate(input)
  user = step :persist, persist(attrs)
  step :notify, notify(user)
  user
end
```

Definitely not for this PR, but I'm keen for your thoughts! Now that we have this result handling, it feels like there is more potential value in (optionally) naming steps like this.

Reply:
I'm also torn here. Not sure we should make this as something built-in in the library as you can already use Ruby + pattern matching for that, so maybe just providing examples would be enough. I mean, we could have the `validate` method return a `[:validation_error, errors]` tuple. I realize that's not too idiomatic, though.

Reply:
Yeah. Let's see about this in the future sometime. We can have a play and see how it feels.